src/ValidatorResult.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-validator for the canonical source repository
+ * @copyright Copyright (c) 2017 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-validator/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\Validator;
+
+/**
+ * Value object representing results of validation.
+ */
+class ValidatorResult implements Result
+{
+    use ValidatorResultMessageInterpolator;
+
+    /**
+     * @var bool
+     */
+    private $isValid;
+
+    /**
+     * Message templates.
+     *
+     * Each may be a self-contained message, or contain placeholders of the form
+     * "%name%" for variables to interpolate into the string.
+     *
+     * @var string[]
+     */
+    private $messageTemplates = [];
+
+    /**
+     * Map of message variable names to the values to interpolate.
+     * @var string[]
+     */
+    private $messageVariables = [];
+
+    /**
+     * @var mixed
+     */
+    private $value;
+
+    /**
+     * @param mixed $value
+     * @param bool $isValid
+     * @param string[] $messageTemplates
+     * @param string[] $messageVariables
+     */
+    public function __construct(
+        $value,
+        bool $isValid,
+        array $messageTemplates = [],
```

Comment:
What are you think to make a collection value object for message templates and variables? We can keep types strictly.